pkg/domain/infra/abi/artifact.go

```diff
@@ -203,11 +203,17 @@ func (ir *ImageEngine) ArtifactAdd(ctx context.Context, name string, artifactBlo
 		return nil, err
 	}
 
+	// If replace is true, try to remove existing artifact (ignore errors if it doesn't exist)
+	if opts.Replace {
+		_, _ = artStore.Remove(ctx, name)
+	}
```

Comment:
my preference would be to examine the error here, and if it is a notexists error, move on; otherwise, blow up OR at the least, drop some debug out.